[rlgl] Compile issue with `GL_HALF_FLOAT_OES` on `GRAPHICS_API_OPENGL_21`

- [x] I tested it on latest raylib version from master branch
 - [x] I checked there is no similar issue already reported
 - [x] I checked the documentation on the [wiki](https://github.com/raysan5/raylib/wiki)
 - [x] My code has no errors or misuse of raylib

### Issue description

While trying to compile the lastest master branch (https://github.com/raysan5/raylib/commit/b82217eaaaabd4540a595958970b78f08cd62844) for `PLATFORM_DESKTOP` with `GRAPHICS_API_OPENGL_21` on Linux I get this error:

```
~/test$ unzip raylib-master.zip
[...]
~/test$ cd raylib-master/src/
~/test/raylib-master/src$ make PLATFORM=PLATFORM_DESKTOP GRAPHICS=GRAPHICS_API_OPENGL_21
gcc -c rcore.c -Wall -D_GNU_SOURCE -DPLATFORM_DESKTOP -DGRAPHICS_API_OPENGL_21 -Wno-missing-braces -Werror=pointer-arith -fno-strict-aliasing  -std=c99 -fPIC -O1 -Werror=implicit-function-declaration -I. -Iexternal/glfw/include -Iexternal/glfw/deps/mingw
In file included from rcore.c:121:
rlgl.h: In function ‘rlGetGlTextureFormats’:
rlgl.h:3270:151: error: ‘GL_HALF_FLOAT_OES’ undeclared (first use in this function); did you mean ‘GL_HALF_FLOAT_ARB’?
 3270 |  RL_PIXELFORMAT_UNCOMPRESSED_R16: if (RLGL.ExtSupported.texFloat16) *glInternalFormat = GL_LUMINANCE; *glFormat = GL_LUMINANCE; *glType = GL_HALF_FLOAT_OES; break;   // NOTE: Requires extension OES_texture_half_float
      |                                                                                                                                           ^~~~~~~~~~~~~~~~~
      |                                                                                                                                           GL_HALF_FLOAT_ARB
rlgl.h:3270:151: note: each undeclared identifier is reported only once for each function it appears in
make: *** [Makefile:687: rcore.o] Error 1
```

The error doesn't happen if I use `GRAPHICS_API_OPENGL_11` or `GRAPHICS_API_OPENGL_33`.

I believe this is related to a previous commit: https://github.com/raysan5/raylib/commit/dc621ca388077a8bd814c8d399dea09efbfc25ce.

@Not-Nik I'm unfamiliar with `GL_HALF_FLOAT_OES`. If you can, could you please take a look at this?

### Environment

Platform: _Desktop_
Operating System: _Linux Mint 21.1 (x86_64)_
OpenGL version: _3.1 Mesa 22.0.5_
GPU: _Intel HD Graphics 3000_

--- a/src/rlgl.h
+++ b/src/rlgl.h
@@ -3267,9 +3267,15 @@ void rlGetGlTextureFormats(int format, unsigned int *glInternalFormat, unsigned 
         case RL_PIXELFORMAT_UNCOMPRESSED_R32: if (RLGL.ExtSupported.texFloat32) *glInternalFormat = GL_LUMINANCE; *glFormat = GL_LUMINANCE; *glType = GL_FLOAT; break;            // NOTE: Requires extension OES_texture_float
         case RL_PIXELFORMAT_UNCOMPRESSED_R32G32B32: if (RLGL.ExtSupported.texFloat32) *glInternalFormat = GL_RGB; *glFormat = GL_RGB; *glType = GL_FLOAT; break;                  // NOTE: Requires extension OES_texture_float
         case RL_PIXELFORMAT_UNCOMPRESSED_R32G32B32A32: if (RLGL.ExtSupported.texFloat32) *glInternalFormat = GL_RGBA; *glFormat = GL_RGBA; *glType = GL_FLOAT; break;             // NOTE: Requires extension OES_texture_float
+        #if defined(GRAPHICS_API_OPENGL_21)
+        case RL_PIXELFORMAT_UNCOMPRESSED_R16: if (RLGL.ExtSupported.texFloat16) *glInternalFormat = GL_LUMINANCE; *glFormat = GL_LUMINANCE; *glType = GL_HALF_FLOAT_ARB; break;
+        case RL_PIXELFORMAT_UNCOMPRESSED_R16G16B16: if (RLGL.ExtSupported.texFloat16) *glInternalFormat = GL_RGB; *glFormat = GL_RGB; *glType = GL_HALF_FLOAT_ARB; break;
+        case RL_PIXELFORMAT_UNCOMPRESSED_R16G16B16A16: if (RLGL.ExtSupported.texFloat16) *glInternalFormat = GL_RGBA; *glFormat = GL_RGBA; *glType = GL_HALF_FLOAT_ARB; break;
+        #else // defined(GRAPHICS_API_OPENGL_ES2)
         case RL_PIXELFORMAT_UNCOMPRESSED_R16: if (RLGL.ExtSupported.texFloat16) *glInternalFormat = GL_LUMINANCE; *glFormat = GL_LUMINANCE; *glType = GL_HALF_FLOAT_OES; break;   // NOTE: Requires extension OES_texture_half_float
         case RL_PIXELFORMAT_UNCOMPRESSED_R16G16B16: if (RLGL.ExtSupported.texFloat16) *glInternalFormat = GL_RGB; *glFormat = GL_RGB; *glType = GL_HALF_FLOAT_OES; break;         // NOTE: Requires extension OES_texture_half_float
         case RL_PIXELFORMAT_UNCOMPRESSED_R16G16B16A16: if (RLGL.ExtSupported.texFloat16) *glInternalFormat = GL_RGBA; *glFormat = GL_RGBA; *glType = GL_HALF_FLOAT_OES; break;    // NOTE: Requires extension OES_texture_half_float
+        #endif
         #endif
         #endif
     #elif defined(GRAPHICS_API_OPENGL_33)

